lib/Service/BoardService.php

```diff
@@ -679,9 +699,92 @@ public function clone($id, $userId) {
 			$this->stackMapper->insert($newStack);
 		}
 
+		if ($withCards) {
+			$this->cloneCards($board, $newBoard, $withAssignments, $withLabels, $withDueDate, $moveCardsToLeftStack, $restoreArchivedCards);
+		}
+
 		return $newBoard;
 	}
 
+	private function cloneCards(Board $board, Board $newBoard, $withAssignments = false, $withLabels = false, $withDueDate = false, $moveCardsToLeftStack = false, $restoreArchivedCards = false) {
```

Comment:
We can put this Card cloning logic into the `CardMapper` class by creating a static method.